### PR TITLE
git credential manager: update name/links

### LINKED ISSUE #12150: git credential manager: update name/links
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/get-started/getting-started-with-git

### What part(s) of the article would you like to see updated?

I'd like to update the Git Credential Manager Core section in Caching Credentials with the new org/product name:

https://docs.github.com/en/get-started/getting-started-with-git/caching-your-github-credentials-in-git#git-credential-manager-core

Git Credential Manager Core --> Git Credential Manager
GCM Core --> GCM
microsoft/Git-Credential-Manager-Core --> GitCredentialManager/git-credential-manager

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ldennington**
### Why:

Closes #12150

The Git Fundamentals team has changed the name of the Git Credential Manager Core tool to Git Credential Manager. The team has also migrated the repository into [its own org](https://github.com/GitCredentialManager). Updating the documentation in this repo to reflect these changes.

### What's being changed:

Git Credential Manager Core --> Git Credential Manager
GCM Core --> GCM
microsoft/Git-Credential-Manager-Core --> GitCredentialManager/git-credential-manager

### Check off the following:

- [X] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@ldennington 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: data/release-notes/enterprise-server/3-2/0-rc1.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ sections:
         - |
           Git Credential Manager (GCM) Core versions 2.0.452 and later now provide security-hardened credential storage and multi-factor authentication support for {% data variables.product.product_name %}.
```
Comment (vgrl) [inline data/release-notes/enterprise-server/3-2/0-rc1.yml:62]: I suggest we remove the reference here also.
```suggestion
          Git Credential Manager (GCM) versions 2.0.452 and later now provide security-hardened credential storage and multi-factor authentication support for {% data variables.product.product_name %}.
```

File: data/release-notes/enterprise-server/3-2/0.yml [hunk 2]
```diff
@@ -59,7 +59,7 @@ sections:
         - |
           Git Credential Manager (GCM) Core versions 2.0.452 and later now provide security-hardened credential storage and multi-factor authentication support for {% data variables.product.product_name %}.
```
Comment (vgrl) [inline data/release-notes/enterprise-server/3-2/0.yml:60]: Same as above, let's drop the reference here too.
```suggestion
          Git Credential Manager (GCM) versions 2.0.452 and later now provide security-hardened credential storage and multi-factor authentication support for {% data variables.product.product_name %}.
```

**Review by vgrl — DISMISSED** [review]
Looks great! Thanks so much for opening a PR with these changes! I'll update the branch and get this merged once tests are passing. 🍏

**Review by vgrl — APPROVED** [review]
👍

**Pull request merged**